src/ds5/ds5-options.h

```diff
@@ -189,41 +189,17 @@ namespace librealsense
     class external_sync_mode : public option
     {
     public:
-        external_sync_mode(hw_monitor& hwm);
+        external_sync_mode(hw_monitor& hwm, sensor_base* depth_ep = nullptr, int ver = 1); // ver = 1, for firmware 5.9.15.1 and later, INTERCAM_SYNC_MAX is 2 with master and slave mode only.
+                                                                                           // ver = 2, for firmware 5.12.4.0 and later, INTERCAM_SYNC_MAX is 258 by adding FULL SLAVE mode and genlock with trigger frequency 1 - 255.
+                                                                                           // ver = 3  for firmware 5.12.12.100 and later, INTERCAM_SYNC_MAX is 260 by adding genlock with laser on-off and Off-On two frames.
+
         virtual ~external_sync_mode() = default;
         virtual void set(float value) override;
         virtual float query() const override;
         virtual option_range get_range() const override;
         virtual bool is_enabled() const override { return true; }
```

Comment:
Override this with
```virtual bool is_enabled() const override { return _sensor && !_sensor ->is_streaming(); }```

as with the previous _v2 version.  It is required to avoid getting the exception by checking `sensor.supports(sync)` first

Reply:
Added. Thanks!